okhttp/src/test/java/okhttp3/internal/connection/ConnectionPoolTest.kt

```diff
@@ -192,6 +200,47 @@ class ConnectionPoolTest {
     assertThat(realTaskRunner.activeQueues()).isEmpty()
   }
 
+  @Test fun connectionPreWarming() {
+    val address = routePlanner.address
+    val pool = factory.newConnectionPool(routePlanner = routePlanner)
+
+    // Set the policy, wait for connection to be created
+    routePlanner.addPlan()
+    pool.setPolicy(address, ConnectionPool.AddressPolicy(1))
+    // TODO why does runTasks() spin forever if I don't first call runNextTask() twice?
+    factory.taskFaker.runNextTask()
+    factory.taskFaker.runNextTask()
+    factory.taskFaker.runTasks()
```

Comment:
Could use some guidance on this

Reply:
Sounds like a bug, Jesse might know.  Otherwise will require debugging.